__tests__/app.test.js

```diff
@@ -16,51 +16,145 @@ beforeEach(() => {
+        }
+      };
+      return request(app)
+        .get("/api/articles/1")
+        .then(({ body }) => {
+          expect(body).toMatchObject(articleTemplate);
         });
-        test("Should return an array with length equal to total number of topics in database", () => {
-            return request(app)
-                .get("/api/topics")
-                .expect(200)
-                .then(({ body }) => {
-                    expect(body.length).toBe(3);
-                });
+    });
+    test("Should return status 404 and error message if article_id does not exist", () => {
+      return request(app)
+        .get("/api/articles/999")
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`We can't find that!`);
         });
-        test(`Each object in the  returned array should have keys of 'slug' and 'description'`, () => {
-            return request(app)
-                .get("/api/topics")
-                .expect(200)
-                .then(({ body }) => {
-                    body.forEach((topic) => {
-                        expect(topic).toHaveProperty("slug");
-                        expect(topic).toHaveProperty("description");
-                    });
-                });
+    });
+    test("Should return status 400 and error message if article_id is invalid type", () => {
+      return request(app)
+        .get("/api/articles/bananaPancakes")
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`That input is invalid`);
         });
-        test("Each object in the returned array should only have 2 keys", () => {
-            return request(app)
-                .get("/api/topics")
-                .expect(200)
-                .then(({ body }) => {
-                    body.forEach((topic) => {
-                        expect(Object.keys(topic).length).toBe(2);
-                    });
-                });
+    });
+  });
+  describe("GET /api/articles/:article_id/comments", () => {
+    test("Should return 200 status code", () => {
+      return request(app).get("/api/articles/1/comments")
+        .expect(200);
+    });
+    test("Should return an array of comment objects with desired keys", () => {
+      const expectedShape = {
+        comment_id: 10,
+        body: "git push origin master",
+        votes: 0,
+        author: "icellusedkars",
+        article_id: 3,
+        created_at: "2020-06-20T07:24:00.000Z",
+      };
+      return request(app)
+        .get("/api/articles/3/comments")
+        .then(({ body }) => {
+          expect(body[1]).toMatchObject(expectedShape)
+        })
+    });
+    test("Should serve comments in order of most recent first", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .then(({ body }) => {
+          expect(body[0].comment_id).toBe(5);
+          expect(body[10].comment_id).toBe(9);
```

Comment:
There is a great jest matcher called toBeSotedBy that can be installed for this test. It makes testing for sorted by ascending or descending a little clearer in your code. Here is an example
.toBeSortedBy('created_at', {descending: true})

Reply:
I gave up trying to get the jest docs to load yesterday (bad internet after powercut)! I had all the syntax right but hadn't put the  extra line inside the package.json so nothing was happening - got it sorted now thanks :)